wasm.md

```diff
@@ -1531,6 +1496,238 @@ Start Function
+ // --------------------------------------------
+    rule <k> ( table                  EXPO:InlineExport SPEC:TableSpec )
+          => ( table #freshId(NEXTID) EXPO              SPEC           )
+          ...
+         </k>
+         <nextFreshId> NEXTID => NEXTID +Int 1 </nextFreshId>
+
+    rule <k> ( table ID:Identifier ( export ENAME ) SPEC:TableSpec )
+          => ( export ENAME ( table ID ) )
+          ~> ( table ID SPEC )
+          ...
+         </k>
+
+    syntax MemorySpec ::= InlineExport MemorySpec
+ // ---------------------------------------------
+    rule <k> ( memory                  EXPO:InlineExport SPEC:MemorySpec )
+          => ( memory #freshId(NEXTID) EXPO              SPEC            )
+          ...
+         </k>
+         <nextFreshId> NEXTID => NEXTID +Int 1 </nextFreshId>
+
+    rule <k> ( memory ID:Identifier ( export ENAME ) SPEC:MemorySpec )
+          => ( export ENAME ( memory ID ) )
+          ~> ( memory ID SPEC )
+          ...
+         </k>
+```
+
+Imports
+-------
+
+Imports need to describe the type of what is imported.
+That an import is really a subtype of the declared import needs to be checked at instantiation time.
+The value of a global gets copied when it is imported.
+
+```k
+    syntax Defn       ::= ImportDefn
+    syntax ImportDefn ::= "(" "import" String String ImportDesc ")"
+    syntax ImportDesc ::= "(" "func"   OptionalId TypeUse        ")" [klabel(funcImportDesc)]
+                        | "(" "table"  OptionalId TableType      ")" [klabel( tabImportDesc)]
+                        | "(" "memory" OptionalId MemType        ")" [klabel( memImportDesc)]
+                        | "(" "global" OptionalId TextGlobalType ")" [klabel(globImportDesc)]
+ // -------------------------------------------------------------------------------------
+    rule <k> ( import MOD NAME (func OID:OptionalId TUSE:TypeUse)) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <typeIds> TYPEIDS </typeIds>
+           <types>   TYPES   </types>
+           <funcIds> IDS => #saveId(IDS, OID, NEXT) </funcIds>
+           <funcIndices> FS => FS [NEXT <- ADDR] </funcIndices>
+           <nextFuncIdx> NEXT => NEXT +Int 1 </nextFuncIdx>
+           ...
+         </moduleInst>
+         <moduleRegistry> ... MOD |-> MODIDX ... </moduleRegistry>
+         <moduleInst>
+           <modIdx> MODIDX </modIdx>
+           <funcIds> IDS' </funcIds>
+           <funcIndices> ... #ContextLookup(IDS' , TFIDX) |-> ADDR ... </funcIndices>
+           <exports>     ... NAME |-> TFIDX                        ... </exports>
+           ...
+         </moduleInst>
+         <funcDef>
+           <fAddr> ADDR </fAddr>
+           <fType> FTYPE </fType>
+           ...
+         </funcDef>
+      requires unnameFuncType(FTYPE) ==K unnameFuncType(asFuncType(TYPEIDS, TYPES, TUSE))
+
+    rule <k> ( import MOD NAME (table OID:OptionalId (LIM _):TableType)) => . ... </k>
```

Comment:
same. you have space after first paren but not before the last paren.

Reply:
Fixed